File: frontend/src/apis/mcp.ts
import { request } from '@/utils';

export interface MCPTool {
  name: string;
  description: string;
}

export const fetchMCPTools = (url: string) => {
  return request.post<{ tools: MCPTool[] }>('/mcp/tools/fetch', { url }, { timeout: 120000 });
};

File: backend/internal/appdto/mcp.go
package appdto

type FetchMCPToolsReq struct {
	URL       string `json:"url" binding:"required"`
	Transport string `json:"transport"` // "sse" or "http"
}

type MCPTool struct {
	Name        string `json:"name"`
	Description string `json:"description"`
}

type FetchMCPToolsResp struct {
	Tools []MCPTool `json:"tools"`
}
